### PR TITLE
Issue #811 Expose logging flag for minishift

### PR DESCRIPTION
**Pull request by praveenkumar**
This patch will add logging flag as part of minishift start.

Addresses issue #811

### CODE REVIEW

File: cmd/minishift/cmd/start.go [hunk 1]
```diff
@@ -353,6 +353,7 @@ func initClusterUpFlags() {
 	clusterUpFlagSet.Int(serverLogLevel, 0, "Log level for the OpenShift server.")
 	clusterUpFlagSet.StringSliceVarP(&openShiftEnv, openshiftEnv, "e", []string{}, "Specify key-value pairs of environment variables to set on the OpenShift container.")
 	clusterUpFlagSet.Bool(metrics, false, "Install metrics (experimental)")
+	clusterUpFlagSet.Bool(logging, false, "Install logging (experimental)")
```
Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: So what type of logging are we talking about here? This is the OpenShift integrated logging, right? (similar to metrics)? This is not logging in terms of CLI logging and stands in no relation to glog or the global log level, right?

File: cmd/minishift/cmd/start.go [hunk 2]
```diff
@@ -239,8 +239,7 @@ func applyAddOns(ip string, routingSuffix string, ocPath string, kubeConfigPath
 	addOnManager := addon.GetAddOnManager()
 	err := addOnManager.Apply(addon.GetExecutionContext(ip, routingSuffix, ocPath, kubeConfigPath, sshCommander))
 	if err != nil {
-		fmt.Println("Error executing addon commands: ", err)
-		atexit.Exit(1)
+		atexit.ExitWithMessage(1, fmt.Sprint("Error executing addon commands: ", err))
```
Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: Could you either squash this commit or use the same issue prefix in the commit message indicating in which context we've done this. Either works for me.

File: cmd/minishift/cmd/start.go (hunk 1 above)
Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: > This is the OpenShift integrated logging, right? (similar to metrics)?

Right it's integrated one.

File: cmd/minishift/cmd/start.go (hunk 2 above)
Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: @hferentschik done, Added issue prefix in the commit message.

File: cmd/minishift/cmd/start.go (hunk 1 above)
Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: So where or what do I look at after the bootstrap to see whether this works?

Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: @hferentschik when you start it with --logging then it should start `kibana` container for log aggregation and also update master-config.yml for update url like below.

- https://docs.openshift.com/container-platform/3.3/install_config/aggregate_logging.html#aggregate-logging-kibana

```
  extensions: null
  loggingPublicURL: https://kibana-logging.192.168.42.244.nip.io
```

Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: @praveenkumar thanks, that's what I was looking for. I was not sure how to access Kibana. Trying ....

Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: @praveenkumar I am getting the infamous "Application is not available" error as with metrics. Does logging deployment work for you? Does logging have the same deployment issues then metrics?

Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: @hferentschik working for me, below are the containers are up and running.

```
docker@minishift:~$ docker ps
CONTAINER ID        IMAGE                                           COMMAND                  CREATED             STATUS              PORTS               NAMES
dc253875ebad        openshift/origin-logging-auth-proxy:v1.5.0      "openshift-auth-proxy"   11 minutes ago      Up 11 minutes                           k8s_kibana-proxy.e1e629b2_logging-kibana-1-l7ghm_logging_efb547b6-2fea-11e7-8776-4e6488e00d24_1613f7fd
8826e8efb65a        openshift/origin-logging-kibana:v1.5.0          "container-entrypoint"   12 minutes ago      Up 12 minutes                           k8s_kibana.b259ec84_logging-kibana-1-l7ghm_logging_efb547b6-2fea-11e7-8776-4e6488e00d24_c6a11932
578c1d3f0324        openshift/origin-logging-elasticsearch:v1.5.0   "sh /opt/app-root/src"   12 minutes ago      Up 12 minutes                           k8s_elasticsearch.4c914266_logging-es-rbb8ayeq-1-6sf1k_logging_edd9dbc6-2fea-11e7-8776-4e6488e00d24_a7a779a3
d06cd16c27b4        openshift/origin-logging-fluentd:v1.5.0         "sh run.sh"              13 minutes ago      Up 13 minutes                           k8s_fluentd-elasticsearch.3c6220ce_logging-fluentd-wc30t_logging_e9dadb4b-2fea-11e7-8776-4e6488e00d24_b64033c9
b1343921e3ce        openshift/origin-logging-curator:v1.5.0         "sh run.sh"              13 minutes ago      Up 13 minutes                           k8s_curator.c045d094_logging-curator-1-9xhq0_logging_ecf6fc10-2fea-11e7-8776-4e6488e00d24_a4ac4422
67807d215040        openshift/origin-pod:v1.5.0                     "/pod"                   13 minutes ago      Up 13 minutes                           k8s_POD.c78901a3_logging-kibana-1-l7ghm_logging_efb547b6-2fea-11e7-8776-4e6488e00d24_b38e3241
a0d9532eb684        openshift/origin-pod:v1.5.0                     "/pod"                   13 minutes ago      Up 13 minutes                           k8s_POD.49de27a2_logging-es-rbb8ayeq-1-6sf1k_logging_edd9dbc6-2fea-11e7-8776-4e6488e00d24_ecf8a48e
f3564f251424        openshift/origin-pod:v1.5.0                     "/pod"                   13 minutes ago      Up 13 minutes                           k8s_POD.4ebadca1_logging-curator-1-9xhq0_logging_ecf6fc10-2fea-11e7-8776-4e6488e00d24_b2bc9988
ce52d859b028        openshift/origin-pod:v1.5.0                     "/pod"                   13 minutes ago      Up 13 minutes                           k8s_POD.4ebadca1_logging-fluentd-wc30t_logging_e9dadb4b-2fea-11e7-8776-4e6488e00d24_484ba4b6
335e35da6fec        openshift/origin-haproxy-router:v1.5.0          "/usr/bin/openshift-r"   14 minutes ago      Up 14 minutes                           k8s_router.a7953138_router-1-xn0ft_default_be561423-2fea-11e7-8776-4e6488e00d24_263268c1
6255d1b48d06        openshift/origin-docker-registry:v1.5.0         "/bin/sh -c 'DOCKER_R"   14 minutes ago      Up 14 minutes                           k8s_registry.6d97d433_docker-registry-1-b78qg_default_c01615af-2fea-11e7-8776-4e6488e00d24_7a312f0c
728d69ee1b95        openshift/origin-pod:v1.5.0                     "/pod"                   14 minutes ago      Up 14 minutes                           k8s_POD.a4a0fe84_docker-registry-1-b78qg_default_c01615af-2fea-11e7-8776-4e6488e00d24_c7b9017d
40379dbb2651        openshift/origin-pod:v1.5.0                     "/pod"                   14 minutes ago      Up 14 minutes                           k8s_POD.4ebadca1_router-1-xn0ft_default_be561423-2fea-11e7-8776-4e6488e00d24_f13b27d9
c47728bb5efe        openshift/origin:v1.5.0                         "/usr/bin/openshift s"   15 minutes ago      Up 15 minutes                           origin

$ curl -Ik https://kibana-logging.192.168.42.183.nip.io
HTTP/1.1 302 Moved Temporarily
X-Powered-By: Express
Location: https://192.168.42.183:8443/oauth/authorize?response_type=code&redirect_uri=https%3A%2F%2Fkibana-logging.192.168.42.183.nip.io%2Fauth%2Fopenshift%2Fcallback&scope=user%3Ainfo%20user%3Acheck-access%20user%3Alist-projects&client_id=kibana-proxy
Content-Length: 0
Set-Cookie: openshift-auth-proxy-session=nXRBnmHWnV6H3tVciLIq_w.4JR3bg_WIAF6YT4BIVxHa0hqhtBveyQNHBnK5e3aO4qDQDfLS46KUWkdmPtWGBX-rG2LSa8NgJxGDNGsHwafCg.1493807974265.3600000.StwpS9BYJ1jHjVlYdqp4c72ZwzK8iMxnRoz77WplV3w; path=/; secure; httponly
Date: Wed, 03 May 2017 10:39:34 GMT
Set-Cookie: c0a2a7a3ed7718ba6090f8aa912a686f=a243509b7f4c53df1cbc4d6736141065; path=/; HttpOnly; Secure
```

Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: And you can also actually view Kabana in the browser?

**Comment by praveenkumar** [comment]
@hferentschik what are we waiting for this PR? I know that there is issue in deployer same as metrics but that doesn't stop to in this PR because this is only to enable flag and flag actually say it's experimental. I need to rebase it though.

**Comment by hferentschik** [comment]
>  what are we waiting for this PR

Nothing. Rebased and merged. Thanks.

**Pull request closed without merging**